javascript/src/GherkinDocumentWalker.ts

```diff
@@ -201,8 +201,8 @@ export default class GherkinDocumentWalker {
 
     this.handlers.handleRule(rule)
 
-    const backgroundKept = children.find((child) => child !== null && child.background !== null)
-    const scenariosKept = children.filter((child) => child !== null && child.scenario !== null)
+    const backgroundKept = children.find((child) => child !== null && child.background)
```

Comment:
I know you're not an expert, but: any thoughts on whether we should relax the `child !== null` comparison in a similar way?

If `child` is always an object, then it's probably a better idea to check for truthiness (is not null or undefined) than to explicitly check for null, even if `child` is not ever expected to take an `undefined` value. (After all, `child.background` and `child.scenario` were expected to never take `undefined`, yet here we now are...)